app/public/cantusdata/helpers/search_utils.py

```diff
@@ -1,82 +1,67 @@
-import re
+# Contains the words that are allowed
+# in a neume_name query
+VALID_NEUME_NAME_WORDS = {
+    "punctum",
+    "pes",
+    "clivis",
+    "scandicus",
+    "torculus",
+    "porrectus",
+    "distropha",
+    "tristopha",
+    "pressus",
+    "climacus",
+    "resupinus",
+    "flexus",
+    "subpunctis",
+    "compound",
+}
 
 
-def get_transpositions(sequence):
-    """Given a series of pitch names (no flats or sharps - just abcdefg),
-    return a list of the 7 possible transpositions of the melody. This is used
-    when generating an elastic search query to look for all transpositions of a
-    user specified pitch sequence.
+def validate_query(q: list[str], q_type: str) -> bool:
+    """
+    Depending on the type of the query, returns True if the query is valid
+    """
+    match q_type:
+        case "neume_names":
+            return all(neume in VALID_NEUME_NAME_WORDS for neume in q)
+        case "pitch_names" | "pitch_names_invariant":
```

Comment:
Ok, I made changed this so that for the backend we have `pitch_names` and `pitch_names_transposed`...we can make the user-facing designation slightly different, but I think that this is probably the simplest for development?